yoti_python_sdk/date_parser.py

```diff
@@ -9,4 +14,29 @@ def from_iso_format(string):
     if len(parts) != 3:
         raise ValueError
 
-    return datetime.date(parts[0], parts[1], parts[2])
+    return date(parts[0], parts[1], parts[2])
+
+
+def datetime_with_microsecond(string):
+    # Python2 does not have a way of parsing date formats.
+    # TODO Deprecate this once Python2 support is dropped.
+    time_split = re.split("[^0-9]", string)
+    parts = len(time_split)
+    if parts <= 6:
+        if logging.getLogger().propagate:
+            logging.warning(ERROR_PARSING_DATE)
+        return None
+
+    year = int(time_split[0]) if parts > 0 else 0
```

Comment:
Not sure if I'm keen on how this is structured - surely we don't need to check for parts <= 6 here if we're checking on line 25? As long as we have a test covering that scenario, any other checking is just adding noise

Reply:
It's because the above check only logs a warning - it doesn't hard fail.